src/layer/x86/deformableconv2d_x86.cpp

```diff
@@ -0,0 +1,248 @@
+// Tencent is pleased to support the open source community by making ncnn available.
+//
+// Copyright (C) 2017 THL A29 Limited, a Tencent company. All rights reserved.
+//
+// Licensed under the BSD 3-Clause License (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// https://opensource.org/licenses/BSD-3-Clause
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include "deformableconv2d_x86.h"
+
+#include "layer_type.h"
+
+namespace ncnn {
+
+DeformableConv2D_x86::DeformableConv2D_x86()
+{
+    one_blob_only = false;
+    support_inplace = false;
+
+    inner_product = 0;
+    permute = 0;
+}
+
+int DeformableConv2D_x86::load_model(const ModelBin& mb)
+{
+    weight_data = mb.load(weight_data_size, 0);
+    if (weight_data.empty())
+        return -100;
+
+    if (bias_term)
+    {
+        bias_data = mb.load(num_output, 1);
+        if (bias_data.empty())
+            return -100;
+    }
+
+    const int in_c = weight_data_size / (num_output * kernel_h * kernel_w);
+    weight_data = weight_data.reshape(kernel_w * kernel_h, in_c, num_output);
+    weight_data_t.create(in_c, kernel_w * kernel_h, num_output);
+    if (weight_data_t.empty())
+        return -100;
+    for (int q = 0; q < num_output; q++)
+    {
+        const Mat m = weight_data.channel(q);
+        float* outptr = weight_data_t.channel(q);
+
+        for (int i = 0; i < kernel_w * kernel_h; i++)
+        {
+            for (int j = 0; j < in_c; j++)
+            {
+                *outptr++ = m.row(j)[i];
+            }
+        }
+    }
+    weight_data_t = weight_data_t.reshape(in_c * kernel_w * kernel_h, num_output);
+    return 0;
+}
```

Comment:
do not override load_model
transform weight data in derived class create_pipeline()